ocaml/testsuite/tests/typing-modes/lazy.ml

```diff
@@ -1,25 +1,134 @@
 (* TEST
+    flags += "-extension unique";
     expect;
 *)
 
-(* lazy expression is legacy *)
-let u =
-    let _x @ portable = lazy "hello" in
-    ()
+let use_portable : 'a @ portable -> unit = fun _ -> ()
 [%%expect{|
-Line 2, characters 24-36:
-2 |     let _x @ portable = lazy "hello" in
-                            ^^^^^^^^^^^^
-Error: This value is nonportable but expected to be portable.
+val use_portable : 'a @ portable -> unit = <fun>
 |}]
 
-(* lazy body is legacy *)
-let x = lazy ("hello" : _ @@ local)
+(* The thunk and the result are required to be global *)
+let foo () =
+    lazy (let x @ local = "hello" in x)
 [%%expect{|
-Line 1, characters 13-35:
-1 | let x = lazy ("hello" : _ @@ local)
-                 ^^^^^^^^^^^^^^^^^^^^^^
+Line 2, characters 37-38:
+2 |     lazy (let x @ local = "hello" in x)
+                                         ^
 Error: This value escapes its region.
+  Hint: It is the result of a lazy expression.
+|}]
+
+let foo (local_ x) =
+    lazy (let _ = x in ())
+[%%expect{|
+Line 2, characters 18-19:
+2 |     lazy (let _ = x in ())
+                      ^
+Error: The value x is local, so cannot be used inside a lazy expression.
+|}]
```

Comment:
Maybe leave a comment saying that the thunk only needs to be global because we don't currently support allocating local values on the stack.

Reply:
Hmm I'm not exactly sure what you mean by that. Do you mean "we don't currently support allocating lazy values on the stack"?

Reply:
Yeah, that's what I meant to write